cost/cloud_cost_anomaly_alerts/cloud_cost_anomaly_alerts.pt

```diff
@@ -196,39 +196,52 @@ script "js_filter_dimensions", type: "javascript" do
       ids: [],
       names: [],
       invalid_ids: [],
-      filters:[]
+      filters: [],
+      expressionsand: [],
+      expressionsor: []
     };
-    var exist = false;
-
-    _.each(param_dimensions, function(user_input_dimension, index) {
-      exist = false;
-      filter = {
-        "dimensionid":"",
-        "dimensionname":"",
-        "type":"",
-        "value":""
-      }
-      dimension_filter_map = user_input_dimension.split("=")
-      hasfilter = false;
-      if (dimension_filter_map.length > 1){
-        hasfilter=true
-      }
-      _.each(ds_dimensions, function(dimension) {
-        if (dimension_filter_map[0] === dimension.name || dimension_filter_map[0] === dimension.id) {
-          filtered_dimensions.ids.push(dimension.id);
-          filtered_dimensions.names.push(dimension.name);
-          exist = true;
-          if (hasfilter) {
-            filter.dimensionid=dimension.id;
-            filter.dimensionname=dimension.name;
-            filter.type="equal";
-            filter.value=dimension_filter_map[1];
-            filtered_dimensions.filters.push(filter);
+    ids = _.map(ds_dimensions, function(value){ return value.id});
+    names = _.map(ds_dimensions, function(value){ return value.name});
+
+    _.each(param_dimensions, function (user_input_dimension) {
+      input_map = user_input_dimension.split("=")
+      if (!(exists = _.contains(ids, input_map[0]) != false ? true : _.contains(names, input_map[0]) != false ? true : false)){
+        filtered_dimensions.invalid_ids.push(input_map[0])
+      }else{
+        var dimensionid = ""
+        var dimensionname = ""
+        if (ids.indexOf(input_map[0]) > -1){
+          element = ids.indexOf(input_map[0])
+          dimensionname = names[element]
+          dimensionid = ids[element]
+          filtered_dimensions.ids.push(dimensionid)
+          filtered_dimensions.names.push(dimensionname)
+        }else {
+          element = names.indexOf(input_map[0])
+          dimensionname = names[element]
+          dimensionid = ids[element]
+          filtered_dimensions.ids.push(dimensionid)
+          filtered_dimensions.names.push(dimensionname)
+        }
```

Comment:
I think you can optimize a little bit from line 213 to 225, because code in `if` and `else` scopes is almost the same, something like:
```
var dimensionid = ""
var dimensionname = ""
var element = ""

if (ids.indexOf(input_map[0]) > -1){
  element = ids.indexOf(input_map[0])
} else {
  element = names.indexOf(input_map[0])
}

dimensionname = names[element]
dimensionid = ids[element]
filtered_dimensions.ids.push(dimensionid)
filtered_dimensions.names.push(dimensionname)
```